access/validator.go

```diff
@@ -421,7 +421,7 @@ func (v *TransactionValidator) checkSufficientBalanceToPayForTransaction(ctx con
 	}
 
 	// return no error if payer has sufficient balance
-	if bool(canExecuteTransaction) {
+	if canExecuteTransaction {
```

Comment:
I think the cast was needed. `DecodeVerifyPayerBalanceResult` returns a `cadence.Bool` not a native `bool`